### PR TITLE
More backticks/refs for manual and add docs about references

### PR DESCRIPTION
**Pull request by kshyatt**

### CODE REVIEW

File: doc/src/manual/documentation.md [hunk 1]
```diff
@@ -585,6 +585,22 @@ parentheses, `( )`, is the URL.
...
+    eigvals!(A,[irange,][vl,][vu]) -> values
+
+Same as [`eigvals`](@ref), but saves space by overwriting the input `A`, instead of creating a copy.
+"""
+```
+
+This will create a link in the generated docs to the `eigvals` documentation
+(which has more information about what this function actually does). It's good to include
+cross references to mutating/non-mutating versions of a function, or to highlight a difference
+between two similar-seeming functions.
```
Comment (MichaelHatherly) [inline doc/src/manual/documentation.md]: Should make it clear here that `@ref` is not markdown syntax, and requires use of Documenter. (Might better to mention this kind of thing in CONTRIBUTING.md instead?)

Comment (kshyatt) [inline doc/src/manual/documentation.md]: Why not both? If I were looking for "how to document Julia code and add cross-refs" I'd look at the manual too.

File: doc/src/manual/documentation.md [hunk 2]
```diff
@@ -585,6 +585,27 @@ parentheses, `( )`, is the URL.
 A paragraph containing a link to [Julia](http://www.julialang.org).
 ```
 
+It's also possible to add cross-references to other documented functions/methdods/variables within
```
Comment (tkelman) [inline doc/src/manual/documentation.md]: methods misspelled

Comment (kshyatt) [inline doc/src/manual/documentation.md]: Nice job, self

**Comment by kshyatt** [comment]
Good to go if CI passes?

File: doc/src/manual/documentation.md [hunk 3]
```diff
@@ -585,6 +585,27 @@ parentheses, `( )`, is the URL.
...
+"""
+```
+
+This will create a link in the generated docs to the `eigvals` documentation
+(which has more information about what this function actually does). It's good to include
+cross references to mutating/non-mutating versions of a function, or to highlight a difference
+between two similar-seeming functions.
+
+!!! note
+The above cross referencing is *not* a Markdown feature, and relies on
```
Comment (tkelman) [inline doc/src/manual/documentation.md]: does the body of this need indenting?

Comment (kshyatt) [inline doc/src/manual/documentation.md]: Yep.

File: doc/src/manual/modules.md [hunk 4]
```diff
@@ -357,26 +357,26 @@ Other known potential failure scenarios include:
...
 
 A few other points to be aware of:
 
 1. No code reload / cache invalidation is performed after changes are made to the source files themselves,
-   (including by `Pkg.update`), and no cleanup is done after `Pkg.rm`
+   (including by [`Pkg.update`](@ref)), and no cleanup is done after [`Pkg.rm`](@ref)
 2. The memory sharing behavior of a reshaped array is disregarded by precompilation (each view gets
    its own copy)
-3. Expecting the filesystem to be unchanged between compile-time and runtime e.g. `@__FILE__`/`source_path()`
+3. Expecting the filesystem to be unchanged between compile-time and runtime e.g. [`@__FILE__`](@ref)/[`source_path()`](@ref)
```
Comment (MichaelHatherly) [inline doc/src/manual/modules.md]: `source_path` isn't exported (see https://travis-ci.org/JuliaLang/julia/jobs/186485491#L771) so will need to be written as `` [`Base.source_path`](@ref) `` to be visible to Documenter.

**Review by MichaelHatherly — CHANGES_REQUESTED** [review]

File: doc/src/manual/modules.md (hunk 4 above)
Comment (kshyatt) [inline doc/src/manual/modules.md]: I think I got it this time?

Comment (MichaelHatherly) [inline doc/src/manual/modules.md]: Actually, `source_path` isn't documented at all as far as I can see, so we can't have a reference to it.

**Comment by kshyatt** [comment]
@MichaelHatherly Is this ok to go?

**Review by MichaelHatherly — APPROVED** [review]

**Pull request merged**